lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,144 @@
+from openedx.core.djangoapps.cors_csrf.authentication import SessionAuthenticationCrossDomainCsrf
+from lms.djangoapps.support.decorators import require_support_permission
+
+
+class EntitlementSupportView(viewsets.ModelViewSet):
+    """
+    Allows viewing and changing learner course entitlements, used the support team.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthenticationCrossDomainCsrf,)
+    permission_classes = (permissions.IsAuthenticated, IsAdminOrAuthenticatedReadOnly,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = SupportCourseEntitlementSerializer
+
+    def filter_queryset(self, queryset):
+        try:
+            username_or_email = self.request.GET.get('user')
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            queryset = queryset.filter(user=user)
+            return super(EntitlementSupportView, self).filter_queryset(queryset).order_by('created')
+        except (KeyError, User.DoesNotExist):
+            return queryset.order_by('created')
+
+    @method_decorator(require_support_permission)
+    def update(self, request):
+        """ Allows support staff to unexpire a user's entitlement."""
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            course_uuid = request.data['course_uuid']
+            reason = request.data['reason']
+            comments = request.data.get('comments', None)
+            entitlement = self.get_most_recent_entitlement(user=user, course_uuid=course_uuid)
+        except KeyError as err:
+            return HttpResponseBadRequest(u'The field {} is required.'.format(text_type(err)))
+        except (User.DoesNotExist, CourseEntitlement.DoesNotExist):
+            return HttpResponseBadRequest(
+                u'Could not find entitlement to course {course_uuid} for user {username}'.format(
+                    course_uuid=course_uuid, username=username_or_email
+                )
+            )
+        if entitlement:
+            if entitlement.expired_at is None:
+                return HttpResponseBadRequest(
+                    u"Entitlement for user {user} to course {course} is not expired.".format(
+                        user=entitlement.user.username,
+                        course=entitlement.course_uuid
+                    )
+                )
+            if entitlement.enrollment_course_run is None:
+                return HttpResponseBadRequest(
+                    u"Entitlement for user {user} to course {course} has not been spent on a course run.".format(
+                        user=entitlement.user.username,
+                        course=entitlement.course_uuid
+                    )
+                )
+            unenrolled_run = self.unexpire_entitlement(entitlement)
```

Comment:
We might want to batch the transactions including the `unexpire_entitlement` method and the creation of the detail log entry with an atomic transaction.  If either action fails we should revert the changes.